### PR TITLE
us-4-task-48

### PR DESCRIPTION
**Pull request by BlayFlorian**

### CODE REVIEW

File: front/src/app/components/form-session/form-session.component.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+@Component({
+  selector: 'app-form-session',
+  templateUrl: './form-session.component.html',
+  styleUrls: ['./form-session.component.scss']
+})
+export class FormSessionComponent implements OnInit {
+
+  @Output() startSession = new EventEmitter<boolean>();
+  form: FormSession = new FormSession();
```
Comment (KevinBoisgontier) [inline front/src/app/components/form-session/form-session.component.ts]: add type to var form

File: front/src/app/pages/home/home.component.ts [hunk 2]
```diff
@@ -9,15 +10,28 @@ import { AuthService } from 'src/app/providers/auth.service';
...
-      this.socket.emit('auth', user);
-    });
-    this.socket.on('message', (ev: any) => {
-      console.log(ev);
+    this.auth.checkAuthentication().then((user: User) => {
+      this.user = user;
     });
   }
 
+  startSession(start: boolean) {
```
Comment (KevinBoisgontier) [inline front/src/app/pages/home/home.component.ts]: Maybe remove the param 'start'

File: front/src/app/pages/home/home.component.ts [hunk 3]
```diff
@@ -9,15 +10,28 @@ import { AuthService } from 'src/app/providers/auth.service';
...
+    if (this.user && start) {
+      this.start = start;
+      this.socket.emit('auth', this.user);
+      this.socket.on('message', (msg: any) => {
+        console.log(msg);
+      });
+    }
+  }
+
+  stop() {
```
Comment (KevinBoisgontier) [inline front/src/app/pages/home/home.component.ts]: Rename in 'stopSession()'

**Comment by osallou** [comment]
@BlayFlorian

General comment (low priority). Always add a catch after *then* to handle errors and display error to user. Else in case of error, javascript will silently fail for user and may have side effects

**Comment by BlayFlorian** [comment]
General comment (WIP).

**Pull request merged**